shell/platform/common/client_wrapper/include/flutter/method_channel.h

```diff
@@ -122,6 +127,41 @@ class MethodChannel {
     messenger_->SetMessageHandler(name_, std::move(binary_handler));
   }
 
+  // Adjusts the number of messages that will get buffered when sending messages
+  // to channels that aren't fully set up yet. For example, the engine isn't
+  // running yet or the channel's message handler isn't set up on the Dart side
+  // yet.
+  //
+  // |new_size] is an int because the deserialization logic handles only 32 bits
+  // values, see
+  // https://github.com/flutter/engine/blob/93e8901490e78c7ba7e319cce4470d9c6478c6dc/lib/ui/channel_buffers.dart#L495.
+  void Resize(int new_size) {
+    auto control_channel = std::make_unique<MethodChannel<EncodableValue>>(
+        messenger_, kControlChannelName, &StandardMethodCodec::GetInstance());
+
+    control_channel->InvokeMethod(
+        kResizeMethod, std::make_unique<EncodableValue>(EncodableList{
+                           EncodableValue(name_),
+                           EncodableValue(new_size),
+                       }));
```

Comment:
Why are these implemented twice, rather than once in a shared location?

Reply:
I moved those methods to a new superclass (same for the public constants above).
Not sure about the name for this class, I came up with ResizableChannel but we might find a better name.

Reply:
Using a superclass was a dead end, I updated the PR using an anonymous namespace instead.